### PR TITLE
feat(KNO-5332): Add heap docs

### PR DESCRIPTION
**Pull request by francoborr**
### Description
Adds docs for the new Heap extension.

### Tasks
[KNO-5332](https://linear.app/knock/issue/KNO-5332/[docs]-add-docs-for-heap-extension)

### CODE REVIEW

File: content/integrations/extensions/heap.mdx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  width="400"
+  src="/images/integrations/extensions/heap_modal.png"
+  alt="Heap extension configuration example"
+/>
+
+## Label custom events
+
+Heap requires the labeling of custom received events from Knock before their use.
+
+1. From you Heap dashboard, navigate to Data > Labeled Events and click on Label event or property > Event.
```
Comment (rachael-t) [inline content/integrations/extensions/heap.mdx:49]: Small typo - "you" to "your"

File: content/integrations/extensions/heap.mdx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  "event": "Notification Sent",
+  "identity": "70171a84-7e52-46d7-8866-5b4ab88cxxx",
+  "properties": {
+    "channelId": "26d3f6ad-eebc-4ce4-9125-bb856dad8xxx",
+    "channelType": "in_app_feed",
+    "environment": "Production",
+    "messageId": "2XoNHQx7lPp0zsCtZh3BGgCJxxx",
+    "provider": "in_app_feed_knock",
+    "stepRef": "YJAxjVC-ya69fBhoSFxxx",
+    "workflowKey": "account-invite-accepted"
```
Comment (rachael-t) [inline content/integrations/extensions/heap.mdx:123]: I noticed looking at the Vercel preview page that there was an extra space before "stepRef" and "workflowKey" but there doesn't appear to one looking at this file 🤔 
Screenshot 2024-02-23 at 1 35 36 PM

File: content/integrations/extensions/heap.mdx (hunk 1 above)
Comment (francoborr) [reply content/integrations/extensions/heap.mdx:49]: Thanks!

File: content/integrations/extensions/heap.mdx (hunk 2 above)
Comment (francoborr) [reply content/integrations/extensions/heap.mdx:123]: Thanks for catching this! It seems that we need to add the `json` prefix in order to correctly parse the JSON. 
I added this to the segments docs as well: https://github.com/knocklabs/docs/pull/409/commits/13f6c06737c6573bcf9a54752325456c1a200169

**Pull request merged**